File: tutorials/LCACifar/scripts/extractCIFAR.m
% This script will unpack the MATLAB version of the CIFAR dataset. Assuming
% the dataset was extracted to OpenPV/tutorials/LCACifar/cifar-10-batches-mat,
% running 'octave extractCIFAR.m' will unpack the images into that directory.

cifarPath = cd(cd("../cifar-10-batches-mat/"));
mixed_file_pathname = [cifarPath, '/mixed_cifar.txt'];

if exist(mixed_file_pathname, "file") == 2
    printf("WARNING: %s exists and is being deleted to prevent duplicate entries.\n", mixed_file_pathname);
    unlink(mixed_file_pathname);
end%if

printf("Extracting Images..");
extractImagesOctave([cifarPath, '/data_batch_1.mat'], 1);
extractImagesOctave([cifarPath, '/data_batch_2.mat'], 2);
extractImagesOctave([cifarPath, '/data_batch_3.mat'], 3);
extractImagesOctave([cifarPath, '/data_batch_4.mat'], 4);
extractImagesOctave([cifarPath, '/data_batch_5.mat'], 5);
extractImagesOctave([cifarPath, '/test_batch.mat'], 0, 0);

File: tutorials/LCACifar/scripts/extractCIFAR.py
#! /usr/bin/env python

import os
import sys
import extractImages

progpath = sys.argv[0]
print(progpath)
progdir = os.path.dirname(progpath)
if len(progdir) == 0:
  progdir = '.'
cifarPath = progdir + '/' + '../cifar-10-batches-mat'
cifarPath = os.path.realpath(cifarPath)
mixed_file_pathname = cifarPath + '/' + 'mixed_cifar.txt'

if os.path.exists(mixed_file_pathname):
  print('WARNING: {} exists and is being deleted to prevent duplicate entries.'.format(mixed_file_pathname));
  os.remove(mixed_file_pathname) 

print('Extracting images..')

extractImages.extractImages(cifarPath + '/' + 'data_batch_1.mat', 1)
extractImages.extractImages(cifarPath + '/' + 'data_batch_2.mat', 2)
extractImages.extractImages(cifarPath + '/' + 'data_batch_3.mat', 3)
extractImages.extractImages(cifarPath + '/' + 'data_batch_4.mat', 4)
extractImages.extractImages(cifarPath + '/' + 'data_batch_5.mat', 5)
extractImages.extractImages(cifarPath + '/' + 'test_batch.mat', 0, False)
